### PR TITLE
SQS Timeout Manager, and lots of asyncio stuff to account for it

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/143

### What changes does this PR make to Grapl? Why?
Port over something important out of rust sqs-executor.
The documentation on it is pretty decent.

### How were these changes tested?
```
INFO:analyzer_executor_lib.sqs_timeout_manager:Loop 1 - waiting 5s for task
INFO:analyzer_executor_lib.analyzer_executor:Sleeping 16
INFO:analyzer_executor_lib.sqs_timeout_manager:Keep alive timed out, informing SQS to raise visibility to 30
INFO:analyzer_executor_lib.sqs_timeout_manager:Loop 2 - waiting 20s for task
INFO:analyzer_executor_lib.analyzer_executor:Slept 16
[...many analyzer_executor things...]
INFO:analyzer_executor_lib.sqs_timeout_manager:Task completed
```

i can probably cook up a test case for it.

### CODE REVIEW

File: src/python/analyzer_executor/src/analyzer_executor_lib/grapl_logger.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""
+Candidate to move to grapl_common
+"""
```
Comment (wimax-grapl) [inline src/python/analyzer_executor/src/analyzer_executor_lib/grapl_logger.py:3]: (fyi)

File: src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+LOGGER = get_module_grapl_logger()
+
+
+class SqsTimeoutManagerException(Exception):
+    pass
+
+
+@dataclass
+class SqsTimeoutManager:
```
Comment (wimax-grapl) [inline src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py:21]: Hello reviewer! I'd highly suggest, before you even look at the code of this class, to **read all the comments first.**

And heck, even read the comments on test_sqs_timeout_manager.py  !

Comment (wimax-grapl) [reply src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py:21]: also compare with https://github.com/grapl-security/grapl/blob/staging/src/rust/sqs-executor/src/sqs_timeout_manager.rs

File: src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+        for i in range(1, self.num_loops + 1):
+            time_to_wait = self._get_next_sleep(i)
+            LOGGER.info(
+                f"SQS MessageID {self.message_id}: Loop {i} - waiting {time_to_wait}s for task"
+            )
+            try:
+                await asyncio.wait_for(
+                    # shield() prevents task from being canceled by the timeout.
+                    asyncio.shield(task),
```
Comment (colin-grapl) [inline src/python/analyzer_executor/src/analyzer_executor_lib/sqs_timeout_manager.py:66]: TIL

**Review by christophermaier — APPROVED** [review]
I'm not much of a Python person yet, but this seems good, for what it's worth.

Documentation comments are much appreciated!

**Pull request merged**